### DISCUSSION TITLE
Have the ability of changing at runtime Geodesic Order to 3 (Vincenty) with Geodesic Precision still double (2) for speed

### DISCUSSION THREAD

**Discussion by soba-noodle** [noelex/GeographicLib.NET#32] [Ideas]
We need to calculate faster than the current (Geodesic order 6 and Geodesic Precision 2) in both Direct and Inverse.   The current one is fixed to Geodesic Precision 2 and Geodesic Order 6 (Karney method) and no way to change them at runtime.   Some location in Geodesic.cs throws exception if we use Geodesic Order 3.   I thought that Geodesic Order 3 (Vincenty method) will speed up considerably (by editing and removing exceptions in the source code) and not much improvements in performance.   We have an old code which implements the Vincenty method which runs very fast (20 times faster than this release).   The change from the current to the option changeable at runtime is not so simple in the current code.   Precision must be double, not float.   It is very nice to have this functionality at runtime.  If we need more accuracy, then we use Karney,  If we need more speed, then we use Vincenty.

**Comment by noelex** [comment]
> I thought that Geodesic Order 3 (Vincenty method) will speed up considerably (by editing and removing exceptions in the source code) and not much improvements in performance.

You'll also need to setup coefficients for `GEOGRAPHICLIB_GEODESIC_ORDER=3` if you want to speed up the calculation.

> Precision must be double, not float.

At the time I created this library, .NET did not support [generic math](https://learn.microsoft.com/en-us/dotnet/standard/generics/math) yet. A lot of work would be required to add generic math support to GeographicLib.NET. I don't see this will happen in the short term.

However, if you just need to do geodesic calculation at lower precision, modifying `Geodesic` an `GeodesicLine` to use `float` won't require too much work.
